velox/exec/Unnest.h

```diff
@@ -40,22 +40,30 @@ class Unnest : public Operator {
 
  private:
   // Generate output for 'size' input rows starting from 'start' input row.
+  // The firstRowStartSize is class member `firstRowStartSize_` and the
+  // endRowStartSize is always 0.
   //
   // @param start First input row to include in the output.
   // @param size Number of input rows to include in the output.
   // @param outputSize Pre-computed number of output rows.
+  // @param firstRowEndSize First input row output end range.
+  // @param endRowEndSize End input row output end range.
   RowVectorPtr generateOutput(
       vector_size_t start,
       vector_size_t size,
-      vector_size_t outputSize);
+      vector_size_t outputSize,
+      vector_size_t firstRowEndSize,
```

Comment:
My understanding is that {start, size, firstRowEndSize, endRowEndSize} describe the range of rows to process and specify that first and last row may need to be processed partially. The first row should start processing from 'firstRowEndSize' index. The last may should end processing at 'endRowEndSize'.

If that's the case, perhaps, it will be clearer to put these 4 integers in a struct with a nice clear name and documentation. Then, use that struct in all methods that currently take these 4 integers.

Perhaps, 

```
struct RowRange {
   vector_size_t start;
   vector_size_t size;
   vector_size_t firstRowStart;
   vector_size_t lastRowEnd;
};
```

Reply:
Overall is right, only a small nit I need to clarify, `The first row should start processing from 'firstRowStartSize_' index and end processing from 'firstRowEndSize'`


> The first row should start processing from 'firstRowEndSize' index.

Reply:
@jinchengchenghh I'm not sure I follow. Here is how I'm thinking about this.

We have "fat" rows. Each row has multiple sub-rows. We want to produce a list of sub-rows for a given range of "fat" rows. Before this change we didn't allow to split "fat" rows. All sub-rows from a single "fat" row were produced together. This change allows to split "fat" rows. Hence, it is no longer enough to request [start, end) range of "fat" rows to emit, but we also need to specify which sub-row to start for the first row and which sub-row to end for the last row.

Am I missing something?

Reply:
Yes, you are totally right.